x-pack/plugins/embeddable_enhanced/public/actions/panel_notifications_action.test.ts

```diff
@@ -5,81 +5,113 @@
+  } as EmbeddableApiContext;
+};
 
 describe('PanelNotificationsAction', () => {
   describe('getDisplayName', () => {
     test('returns "0" if embeddable has no events', async () => {
       const context = createContext();
       const action = new PanelNotificationsAction();
 
-      const name = await action.getDisplayName(context);
+      const name = action.getDisplayName(context);
       expect(name).toBe('0');
     });
 
     test('returns "2" if embeddable has two events', async () => {
       const context = createContext([{}, {}]);
       const action = new PanelNotificationsAction();
 
-      const name = await action.getDisplayName(context);
+      const name = action.getDisplayName(context);
       expect(name).toBe('2');
     });
+
+    test('updates display name when dynamic actions is updated', async () => {
+      const context = createContext([{}, {}]);
+      const action = new PanelNotificationsAction();
+
+      (context.embeddable as HasDynamicActions).setDynamicActions({
+        dynamicActions: { events: [{}, {}, {}] as any[] },
+      });
+
+      const name = action.getDisplayName(context);
+      expect(name).toBe('3');
+    });
   });
 
   describe('getDisplayNameTooltip', () => {
     test('returns empty string if embeddable has no event', async () => {
       const context = createContext();
       const action = new PanelNotificationsAction();
 
-      const name = await action.getDisplayNameTooltip(context);
+      const name = action.getDisplayNameTooltip(context);
       expect(name).toBe('');
     });
 
     test('returns "1 drilldown" if embeddable has one event', async () => {
       const context = createContext([{}]);
       const action = new PanelNotificationsAction();
 
-      const name = await action.getDisplayNameTooltip(context);
+      const name = action.getDisplayNameTooltip(context);
       expect(name).toBe('Panel has 1 drilldown');
     });
 
     test('returns "2 drilldowns" if embeddable has two events', async () => {
       const context = createContext([{}, {}]);
       const action = new PanelNotificationsAction();
 
-      const name = await action.getDisplayNameTooltip(context);
+      const name = action.getDisplayNameTooltip(context);
       expect(name).toBe('Panel has 2 drilldowns');
     });
 
     test('returns "3 drilldowns" if embeddable has three events', async () => {
       const context = createContext([{}, {}, {}]);
       const action = new PanelNotificationsAction();
 
-      const name = await action.getDisplayNameTooltip(context);
+      const name = action.getDisplayNameTooltip(context);
       expect(name).toBe('Panel has 3 drilldowns');
     });
+
+    test('updates tooltip when dynamic actions is updated', async () => {
+      const context = createContext([{}, {}, {}]);
+      const action = new PanelNotificationsAction();
+
+      (context.embeddable as HasDynamicActions).setDynamicActions({
+        dynamicActions: { events: [{}, {}] as any[] },
```

Comment:
type `as unknown as SerializedEvent[]`

Reply:
Done in 343e464a979251cae24177f87eb348c52ab673b0